share/spice/doi/doi.js

```diff
@@ -0,0 +1,78 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_doi = function(api_result){
+
+        if (!api_result || !api_result.DOI || !api_result.author || !api_result.title || api_result.error) {
+            return Spice.failed('doi');
+        }
+
+        Spice.add({
+            id: "doi",
+
+            name: "Reference", // Or Science?
+            data: api_result,
+            signal: 'high',
+            meta: {
+                sourceName: "dx.doi.org",
+                sourceUrl: api_result.URL
+            },
+            normalize: function(item) {
+                return {
+                    url: item.URL,
+                    title: item.title,
+                    subtitle: sprintf( '%s, %s, DOI:%s', format_authors(item.author), item.issued.raw, item.DOI ),
```

Comment:
Haha, don't ask me where it comes from - I was thinking "js has sprintf, right? Oh, it does!"